builtin/logical/database/path_roles.go

```diff
@@ -355,20 +322,65 @@ func (b *databaseBackend) pathRoleCreateUpdate(ctx context.Context, req *logical
 		role = &roleEntry{}
 	}
 
-	if err := pathRoleCreateUpdateCommon(ctx, role, req.Operation, data); err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+	createOperation := (req.Operation == logical.CreateOperation)
+
+	// DB Attributes
+	{
+		if dbNameRaw, ok := data.GetOk("db_name"); ok {
+			role.DBName = dbNameRaw.(string)
+		} else if createOperation {
+			role.DBName = data.Get("db_name").(string)
+		}
+		if role.DBName == "" {
+			return logical.ErrorResponse("database name is required"), nil
+		}
+	}
+
+	// Statements
+	{
+		if creationStmtsRaw, ok := data.GetOk("creation_statements"); ok {
+			role.Statements.Creation = creationStmtsRaw.([]string)
+		} else if createOperation {
+			role.Statements.Creation = data.Get("creation_statements").([]string)
+		}
+
+		if revocationStmtsRaw, ok := data.GetOk("revocation_statements"); ok {
+			role.Statements.Revocation = revocationStmtsRaw.([]string)
+		} else if createOperation {
+			role.Statements.Revocation = data.Get("revocation_statements").([]string)
+		}
+
+		if rollbackStmtsRaw, ok := data.GetOk("rollback_statements"); ok {
+			role.Statements.Rollback = rollbackStmtsRaw.([]string)
+		} else if createOperation {
+			role.Statements.Rollback = data.Get("rollback_statements").([]string)
+		}
+
+		if renewStmtsRaw, ok := data.GetOk("renew_statements"); ok {
+			role.Statements.Renewal = renewStmtsRaw.([]string)
+		} else if createOperation {
+			role.Statements.Renewal = data.Get("renew_statements").([]string)
+		}
+
+		// Do not persist deprecated statements that are populated on role read
+		role.Statements.CreationStatements = ""
+		role.Statements.RevocationStatements = ""
+		role.Statements.RenewStatements = ""
+		role.Statements.RollbackStatements = ""
 	}
 
+	role.Statements.Revocation = strutil.RemoveEmpty(role.Statements.Revocation)
```

Comment:
Just curious why we do this here for only this one.

Reply:
Nevermind! I see this code was just moved from below and has been previously reviewed.

Reply:
Unfortunately I have the same reply as [the other comment about why something is there][1]; I don't know, that's the way it was there before. See pre-#6834:

- https://github.com/hashicorp/vault/blob/8054cc660f582c857736e675201d06c3dcf07888/builtin/logical/database/path_roles.go#L236

[1]: https://github.com/hashicorp/vault/pull/6951#discussion_r296356324